src/common/tusb_fifo.c

```diff
@@ -106,7 +110,7 @@ static void _ff_push_n(tu_fifo_t* f, void const * data, uint16_t n, uint16_t wRe
     memcpy(f->buffer + (wRel * f->item_size), data, nLin*f->item_size);
 
     // Write data wrapped around
-    memcpy(f->buffer, data + nLin*f->item_size, (n - nLin) * f->item_size);
+    memcpy(f->buffer, (uint8_t*)((uintptr_t)data + nLin*f->item_size), (n - nLin) * f->item_size);
```

Comment:
Maybe only `((uint8_t const*) data) + nLin*item_size)` is good enough. I am not entirely sure but `uintptr_t` seems to be optional type for some platforms. Best to keep it as portable as possible